### PR TITLE
Add services

### PR DESCRIPTION
**Pull request by barrachri**
Updated: 
* docker version
* docker-compose version

Added:
* first draft for services and swarm
* tests for both

### CODE REVIEW

File: aiodocker/services.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            "Labels": labels,
+            'TaskTemplate': task_template,
+            'Mode': mode,
+            'Networks': networks,
+            'EndpointSpec': endpoint_spec
+        }
+
+        data = {}
+
+        if config is not None and isinstance(config, dict):
```
Comment (jettify) [inline aiodocker/services.py]: this check is not required since we defined config couple lines earlier

File: aiodocker/services.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            for k, v in config.items():
+                if v is not None:
+                    data[k] = v
+        elif config is not None:
```
Comment (jettify) [inline aiodocker/services.py]: same, this elif can not be reached

File: aiodocker/services.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            'TaskTemplate': task_template,
+            'Mode': mode,
+            'Networks': networks,
+            'EndpointSpec': endpoint_spec
+        }
+
+        data = {}
+
+        if config is not None and isinstance(config, dict):
+            for k, v in config.items():
```
Comment (jettify) [inline aiodocker/services.py]: minor thing, but what do you think about `{k: v for k, v in config.items() if v is not None}` instead for loop ?

File: aiodocker/services.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
+import json
+
+
+class DockerServices(object):
+    def __init__(self, docker):
+        self.docker = docker
+
+    async def list(self, filters: dict={}) -> list:
```
Comment (jettify) [inline aiodocker/services.py]: having mutable structure as dict is not good idea, probable should be 
`filters: Optional[dict]=None` and then `params=filters or {}`

**Comment by barrachri** [comment]
@jettify thanks for the feedback.

I am going to prepare a new PR.

**Comment by barrachri** [comment]
@jettify :)

**Comment by jettify** [comment]
Asked question about typing, in #29 , what is your thoughts on this?

**Comment by barrachri** [comment]
Personally I am 👍 in adding typing. Would be useful also to verify the type of the inputs.

**Comment by barrachri** [comment]
I cleaned the typing part.

**Comment by achimnol** [comment]
All tests pass at my local setup merged with this PR.
Unfortunately I don't have enough time to make detailed code reviews right now.
@jettify Would you like to continue reviewing?

**Comment by asvetlov** [comment]
I have the same note -- please make optional arguments keyword-only.

**Comment by jettify** [comment]
Do we want to change f strings to format here?

File: aiodocker/utils.py [hunk 5]
```diff
@@ -57,3 +60,67 @@ def httpize(d: Optional[dict]) -> Optional[dict]:
     d = decoder.decode(b'', final=True)
     if d:
         yield d
+
+
+def clean_config(config: Optional[dict]) -> dict:
```
Comment (jettify) [inline aiodocker/utils.py]: lets also add tests for this helpers, should be very easy, we can do this in separate PR since, current one is already large.

**Comment by asvetlov** [comment]
Please merge master to make sure your tests are passed on python 3.5

**Comment by asvetlov** [comment]
You clone still conflicts with master :(

File: tests/test_services.py [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+
+    TaskTemplate = {
+        "ContainerSpec": {
+            "Image": "redis",
+            },
+        }
+
+    my_services = []
+    for i in range(3):
+        name = f"service-{i}"
```
Comment (asvetlov) [inline tests/test_services.py]: Tests should go on Python 3.5 too

**Comment by barrachri** [comment]
Ufff, cleaned everything :)

**Comment by asvetlov** [comment]
Cool!
